source/_posts/2023-09-06-release-20239.markdown

```diff
@@ -134,45 +125,104 @@ Other more advanced fields, like attributes, are not available from the UI. You
+```
+
+{% endraw %}
 
-Trigger template entities can now have an action block that is executed after the trigger but before the entities are rendered. All variables from the action block will be available during the entity rendering. This will allow users to fetch things like weather forecast or calendar events and use them in their template entities. Long term goal is to align the trigger template entities with automation syntax so we can support things like tracing too.
 
 ## Weather forecast service
 
-We added a weather forecast service, that returns the weather forecast.
+We added a new service for weather entities, `weather.get_forecast`, that responds with the weather forecast.
 
 Previously you would have to look at the `forecast` attribute of a weather entity to get the forecast, now this can be done by calling a new service, `weather.get_forecast`, that will respond with the forecast.
-You can also specify what type of forecast you want, daily or do you want to know it for every hour? This means you no longer need separate entities for every forecast type, but it can all be done in 1 entity.
-
-If an entity supports multiple forecasts, you can choose which one you want to see in the more info dialog.
-In the weather card in your dashboard, you can also now pick what kind of weather forecast you want to show in that card.
+You can also specify what type of forecast you want, do you want to see the forecast by day or do you want to know it for every hour? This means you no longer need separate entities for every forecast type, but it can all be done in 1 entity.
 
 All weather integrations are updated to support this new service.
 
-With this change, the `forecast` attribute of the weather entity is deprecated, and will be removed in Home Assistant Core 2024.3.
+<p class='img'>
+<img src='/images/blog/2023-09/get-forecast-service.png'>
+Screenshot of service developer tools showing the weather forecast service.
+</p>
 
-You can use the newly added `action` part of a template entity to fetch the forecast and use it in your template.
+If an entity supports multiple forecasts, you can choose which forecast you want to see in the more info dialog.
+In the weather card in your dashboard, you can also now pick what kind of weather forecast you want to show in that card.
 
 <p class='img'>
-<img src='https://media.discordapp.net/attachments/427516175237382144/1146445861421846608/SPOILER_CleanShot_2023-08-30_at_16.03.542x.png?width=1466&height=794'></a>
-Temporary screenshot that needs to be replaced.
+<img src='/images/blog/2023-09/weather-more-info.png'>
+Screenshot of the weather more info dialog.
 </p>
 
+With this change, the `forecast` attribute of the weather entity is deprecated, and will be removed in Home Assistant Core 2024.3.
+This will make it possible to have on-demand forecasts, that update when you want them to, instead of them being updated all time. It will also make the system faster. 
```

Comment:
```suggestion
With this change, the `forecast` attribute of the weather entity is deprecated, and will be removed in Home Assistant Core 2024.3.
This will make it possible to have on-demand forecasts, that update when you want them to, instead of them being updated all time. It will also make the system faster since all state data gets sent to your browser or mobile app every time it is open or woken, and the forecast attribute made weather entities the equivalent of e.g. 100 other entities.
Even with this change, rest assured that the forecast cards in your dashboard will still work, they will just now use the service to retrieve the forecast data asynchronously instead of reading it from the locally stored state.
```

Open to changes here but trying to incorporate some of @bdraco 's explanation from the threads I linked above